tests/posix/common/src/pthread_attr.c

```diff
@@ -0,0 +1,638 @@
+/*
+ * Copyright (c) 2024, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <pthread.h>
+
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+
+#define BIOS_FOOD           0xB105F00D
+#define SCHED_INVALID       4242
+#define INVALID_DETACHSTATE 7373
+
+static bool attr_valid;
+static pthread_attr_t attr;
+static const pthread_attr_t uninit_attr;
+static bool detached_thread_has_finished;
+
+/* TODO: this should be optional */
+#define STATIC_THREAD_STACK_SIZE (MAX(1024, PTHREAD_STACK_MIN + CONFIG_TEST_EXTRA_STACK_SIZE))
+static K_THREAD_STACK_DEFINE(static_thread_stack, STATIC_THREAD_STACK_SIZE);
+
+static void *thread_entry(void *arg)
+{
+	bool joinable = (bool)POINTER_TO_UINT(arg);
+
+	if (!joinable) {
+		detached_thread_has_finished = true;
+	}
+
+	return NULL;
+}
+
+static void create_thread_common(const pthread_attr_t *attrp, bool expect_success, bool joinable)
+{
+	pthread_t th;
+
+	if (!joinable) {
+		detached_thread_has_finished = false;
+	}
+
+	if (expect_success) {
+		zassert_ok(pthread_create(&th, attrp, thread_entry, UINT_TO_POINTER(joinable)));
+	} else {
+		zassert_not_ok(pthread_create(&th, attrp, thread_entry, UINT_TO_POINTER(joinable)));
+		return;
+	}
+
+	if (joinable) {
+		zassert_ok(pthread_join(th, NULL), "failed to join joinable thread");
+		return;
+	}
+
+	/* should not be able to join detached thread */
+	zassert_not_ok(pthread_join(th, NULL));
+
+	for (size_t i = 0; i < 10; ++i) {
+		k_msleep(2 * CONFIG_PTHREAD_RECYCLER_DELAY_MS);
+		if (detached_thread_has_finished) {
+			break;
+		}
+	}
+
+	zassert_true(detached_thread_has_finished, "detached thread did not seem to finish");
+}
+
+static inline void can_create_thread(const pthread_attr_t *attrp)
+{
+	create_thread_common(attrp, true, true);
+}
+
+static inline void cannot_create_thread(const pthread_attr_t *attrp)
+{
+	create_thread_common(attrp, false, true);
+}
+
+ZTEST(pthread_attr, test_null_attr)
+{
+	if (!IS_ENABLED(CONFIG_DYNAMIC_THREAD)) {
+		ztest_test_skip();
+	}
+
+	create_thread_common(NULL, true, true);
+}
+
+ZTEST(posix_apis, test_pthread_attr_static_corner_cases)
+{
+	pthread_attr_t attr1;
+
+	if (IS_ENABLED(CONFIG_DYNAMIC_THREAD)) {
+		ztest_test_skip();
+	}
```

Comment:
you could use `Z_TEST_SKIP_IFDEF` if you really want to

Reply:
Fixed!